identity-service/src/routes/push_notifications.js

```diff
@@ -120,23 +146,26 @@ module.exports = function (app) {
         }
       })
 
+      let deleteUserNotificationSettings = tokenObj.deviceType !== DEVICE_TYPES.SAFARI
+
       if (tokenObj) {
         // delete the endpoint from AWS SNS
-        await deleteEndpoint({ EndpointArn: tokenObj.awsARN })
+        if (tokenObj.awsARN) await deleteEndpoint({ EndpointArn: tokenObj.awsARN })
         await tokenObj.destroy()
         tokenDeleted = true
       }
 
       // delete user
-      const settingsObj = await models.UserNotificationMobileSettings.findOne({
-        where: {
-          userId
+      if (deleteUserNotificationSettings) {
```

Comment:
can you put a comment here explaining why not safari

Reply:
yup